src/DocumentWatcher.ts

```diff
@@ -172,6 +173,22 @@ function applyEditorConfigToTextEditor(
 	/* tslint:disable:no-any */
 	textEditor.options = <any> newOptions;
 	/* tslint:enable */
+
+	applyEndOfLineSetting();
+
+	function applyEndOfLineSetting() {
+		for (const [regex, eol] of new Map([
+			[/^lf$/i, EndOfLine.LF],
+			[/^crlf$/i, EndOfLine.CRLF]
+		])) {
+			if (regex.test(editorconfig.end_of_line)) {
+				textEditor.edit(edit => {
+					edit.setEndOfLine(eol);
+				});
+				return;
+			}
+		}
```

Comment:
Why the `for..of` loop? Might be missing something here but this looks cleaner :).

```js
const EOL = {
    lf: EndOfLine.LF,
    crlf: EndOfLine.CRLF
};

function applyEndOfLineSetting() {
    const endOfLine = EOL[editorconfig.end_of_line];

    if(!endOfLine) {
        return Promise.resolve();
    }

    return textEditor.edit(edit => {
        edit.sendEndOfLine(endOfLine);
    });
}
```

It might be better to create a [transformation](https://github.com/editorconfig/editorconfig-vscode/tree/master/src/transformations) for this?